### PR TITLE
multiple deletion in public group chat

### PR DESCRIPTION
**Pull request by RyanRory**
### First time contributor checklist

- [ ] I have read [how to contribute](https://github.com/signalapp/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [ ] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [ ] I am following the [Code Style Guidelines](https://github.com/signalapp/Signal-Android/wiki/Code-Style-Guidelines)
- [ ] I have tested my contribution on these devices:
 * Device A, Android X.Y.Z
 * Device B, Android Z.Y
 * Virtual device W, Android Y.Y.Z
- [ ] My contribution is fully baked and ready to be merged as is
- [ ] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### CODE REVIEW

File: src/org/thoughtcrime/securesms/conversation/ConversationFragment.java [hunk 1]
```diff
@@ -520,53 +524,97 @@ public void onClick(DialogInterface dialog, int which) {
...
-              }
-
-              if (isThreadDeleted) {
-                threadId = -1;
-                listener.setThreadId(threadId);
+              else {
+                ignoredMessages.add(messageRecord.getId());
               }
             }
+            long[] serverIDsArray = new long[serverIDs.size()];
```
Comment (Mikunj) [inline src/org/thoughtcrime/securesms/conversation/ConversationFragment.java]: With the review changes in `android-service` you wouldn't need to convert `serverIDs` to an array.

File: src/org/thoughtcrime/securesms/conversation/ConversationFragment.java [hunk 2]
```diff
@@ -520,53 +524,97 @@ public void onClick(DialogInterface dialog, int which) {
...
+            for (int i=0; i < serverIDs.size(); i++) {
+              serverIDsArray[i] = serverIDs.get(i);
+            }
+            if (publicChat != null && publicChatAPI != null) {
+              publicChatAPI
+              .deleteMessages(serverIDsArray, publicChat.getChannel(), publicChat.getServer(), isSentByUser)
+              .success(l -> {
+                List<Long> longList = Longs.asList(l);
```
Comment (Mikunj) [inline src/org/thoughtcrime/securesms/conversation/ConversationFragment.java]: Same with this, with the changes in `android-service` we would be able to use `List<Long>` straight from the promise

**Comment by Mikunj** [comment]
Looking good @RyanRory, if you could remove the commented out code and test to see if it works correctly then we can get this merged in :)

File: src/org/thoughtcrime/securesms/conversation/ConversationFragment.java [hunk 3]
```diff
@@ -414,7 +417,8 @@ private void setCorrectMenuVisibility(Menu menu) {
       menu.findItem(R.id.menu_context_reply).setVisible(isPublicChat && selectedMessageCount == 1);
       String userHexEncodedPublicKey = TextSecurePreferences.getLocalNumber(getContext());
       boolean userCanModerate = isPublicChat && LokiPublicChatAPI.Companion.isUserModerator(userHexEncodedPublicKey, publicChat.getChannel(), publicChat.getServer());
-      boolean isDeleteOptionVisible = isPublicChat && selectedMessageCount == 1 && (isSentByUser || userCanModerate);
+//      boolean isDeleteOptionVisible = isPublicChat && selectedMessageCount == 1 && (isSentByUser || userCanModerate);
```
Comment (gmbnt) [inline src/org/thoughtcrime/securesms/conversation/ConversationFragment.java]: @RyanRory you can just delete the old code next time. No need to keep it around :)

File: src/org/thoughtcrime/securesms/conversation/ConversationFragment.java [hunk 4]
```diff
@@ -414,7 +417,8 @@ private void setCorrectMenuVisibility(Menu menu) {
...
+      boolean isDeleteOptionVisible = isPublicChat && (isSentByUser || userCanModerate);
```
Comment (gmbnt) [inline src/org/thoughtcrime/securesms/conversation/ConversationFragment.java]: `isSentByUser` now doesn't make sense anymore. We should either remove the flag or update it to make sense again.

**Pull request merged**